### PR TITLE
Add a simple documented example

### PR DESCRIPTION
**Pull request by psiphi75**
Added a simple documented example on how to use real-world data and how to extract the results.  Only one file was created, this is `examples/simple.rs`.  No other changes were made.

I needed to get my head around how to do this, so I did, and I thought I would provide it.

### CODE REVIEW

**Comment by Norlock** [comment]
Thanks I was already looking for something like this! Since I didn't understand how to use this lib, this is making it really easy.

**Comment by hkolbeck** [comment]
Thank you! A major frustration of mine with DSP is the reluctance to document some usage basics, that this has been sitting untouched this long is itself frustrating.

File: examples/simple.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Do all the planning
+    let mut planner = FftPlanner::<f32>::new();
+    let fft = planner.plan_fft_forward(bins);
+
+    // Do the FFT, this will take the buffer, and write back into to.
+    fft.process(&mut buffer);
+
+    // The fft real results will only be in the first half of the buffer.
+    // This is the Nyquist limit, which is half the sample frequency.
```
Comment (WalterSmuts) [inline examples/simple.rs:52]: I think you might mean:
> The fft *positive* results....

The first half of the buffer still contains imaginary numbers.

File: examples/simple.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    let sample_rate = 11025; // Hz
+
+    // The buffer is used to store the wave data and the FFT result
+    let mut buffer = create_wave(num_samples, sample_rate as f32);
+
+    // Do all the planning
+    let mut planner = FftPlanner::<f32>::new();
+    let fft = planner.plan_fft_forward(bins);
+
+    // Do the FFT, this will take the buffer, and write back into to.
```
Comment (WalterSmuts) [inline examples/simple.rs:48]: > and write back into to.

and write the results back into itself?

File: examples/simple.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    (0..num_samples).into_iter().map(generator).collect()
+}
+
+fn main() {
+    // The number of frequency bins.
+    let bins = 2048;
+
+    // The number of samples to calculate in the sine function.
+    let num_samples = bins * 100;
+    let sample_rate = 11025; // Hz
```
Comment (WalterSmuts) [inline examples/simple.rs:39]: Any reason not to factor this out along with the other constants?

**Pull request closed without merging**